### ISSUE TITLE
cf copy-source can't find space

### ISSUE THREAD

**Issue by dmcintyre-pivotal** [cloudfoundry/cli#849]
Running on Mac OSX 10.11.5.

cf copy-source fails, saying it can't find a space which does exist:

```
MacBook-Pro-4:cf-SpringBootTrader dmcintyre$ cf -v
cf version 6.18.1+a1103f0-2016-05-24
MacBook-Pro-4:cf-SpringBootTrader dmcintyre$ cf target -o demos -s dmcintyre
                
API endpoint:   https://api.my-pcf.local (API version: 2.54.0)
User:           dmcintyre@pivotal.io
Org:            demos
Space:          dmcintyre
MacBook-Pro-4:cf-SpringBootTrader dmcintyre$ cf apps
Getting apps in org demos / space dmcintyre as dmcintyre@pivotal.io...
OK

name      requested state   instances   memory   disk   urls
pcfdemo   started           4/4         1G       1G     pcfdemo-yourname.apps.my-pcf.local
MacBook-Pro-4:cf-SpringBootTrader dmcintyre$ cf target -o dmcintyre -s basic-demo
                
API endpoint:   https://api.my-pcf.local (API version: 2.54.0)
User:           dmcintyre@pivotal.io
Org:            dmcintyre
Space:          basic-demo
MacBook-Pro-4:cf-SpringBootTrader dmcintyre$ cf apps
Getting apps in org dmcintyre / space basic-demo as dmcintyre@pivotal.io...
OK

No apps found
MacBook-Pro-4:cf-SpringBootTrader dmcintyre$ cf target -o demos -s dmcintyre
                
API endpoint:   https://api.my-pcf.local (API version: 2.54.0)
User:           dmcintyre@pivotal.io
Org:            demos
Space:          dmcintyre
MacBook-Pro-4:cf-SpringBootTrader dmcintyre$ cf copy-source pcfdemo pcfdemo -o dmcintyre -s basic-demo
FAILED
Could not find space basic-demo in organization dmcintyre
```

**Comment by dkoper** [comment]
That looks suspicious - we'll take a look.
Note that `cf copy-source` copies from one app to another, so the destination app needs to exist - your output shows no app `pcfdemo` in the destination space. So I expected a different error message.

**Comment by dmcintyre-pivotal** [comment]
Copies from one app to another? Does the app-to-be-replaced need to be of the same type as the source / need to have used the same buildpack?

I thought this was a simple way of moving or copying an app from one space to another without requiring the bits locally. Is there more/less to it than that?

The command help could do with being a bit more complete I think.

**Comment by dkoper** [comment]
The upcoming V3 API should allow us to introduce a command that allows you to copy (or move) a deployed app from one space to another.

Currently, all that's there is this command (using http://apidocs.cloudfoundry.org/237/apps/copy_the_app_bits_for_an_app.html) that copies the app bits (the files that were uploaded by `cf push`, so before staging) to another app.

I believe the destination will be restaged (as it's not the droplet that's being copied but the pre-staging files), hence it could be picked up by another buildpack (assuming the dest app wasn't initially pushed with a particular buildpack specified).
@SocalNick Can you confirm this is correct?

I welcome suggestions on how to improve the command help!

**Comment by SocalNick** [comment]
The documentation linked above seems to be accurate:

This endpoint will copy the package bits in the blobstore from the source app to the destination app. It will always return a job which you can query for success or failure. This operation will require the app to restart in order for the changes to take effect.

**Comment by dkoper** [comment]
@SocalNick From my understanding, it seems it'll do a restage, which is much more than a restart (it certainly takes longer, but could also mean changes to env vars or bound services would be reflected that otherwise (depending on the buildpack) would not be).

Maybe my confusion is with the `This operation will require the app to restart in order for the changes to take effect.` which can be interpreted two ways: the user should restart the app, or the invocation of this operation initiates the restart (as it requires the app to restart)?

**Comment by SocalNick** [comment]
The [Copy the app bits for an App](http://apidocs.cloudfoundry.org/237/apps/copy_the_app_bits_for_an_app.html) endpoint just copies the app bits. Restart in V2 means issue a stop and then a restart, and because the app has new bits, it will be staged. This is what the CLI does (at least in version 6.17.1+2ca614d-2016-05-02).

I think the original issue above might be a slight behavior change in the CLI:

```
○ → cf --version
cf version 6.17.1+2ca614d-2016-05-02
○ → cf copy-source flora bora -o ncalugar-a1 -s foo
FAILED
App bora not found
```

Upgraded my CLI, then:
```
○ → cf --version
cf version 6.18.1+a1103f0-2016-05-24
○ → cf copy-source flora bora -o ncalugar-a1 -s foo
FAILED
Could not find space foo in organization ncalugar-a1
```

**Comment by dkoper** [comment]
This has been fixed in cf CLI 6.20.0.
https://github.com/cloudfoundry/cli#downloads

Thanks for bringing it to our attention!

Regards,
Dies Koper
CF CLI PM

**Comment by smkivuos** [comment]
quick question w.r.t the comment above link : https://github.com/cloudfoundry/cli/issues/849#issuecomment-222844160
 is there any cf api allows us to copy (or move) a deployed app from one space to another space
